packages/alea-frontend/pages/api/job-portal/create-job-application.ts

```diff
@@ -0,0 +1,32 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { checkIfPostOrSetError, executeAndEndSet500OnError, executeQuery } from '../comment-utils';
+import { getUserIdIfAuthorizedOrSetError } from '../access-control/resource-utils';
+import { Action, CURRENT_TERM, ResourceName } from '@stex-react/utils';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!checkIfPostOrSetError(req, res)) return;
+  const userId = await getUserIdIfAuthorizedOrSetError(
+    req,
+    res,
+    ResourceName.JOB_PORTAL,
+    Action.APPLY,
+    { instanceId: CURRENT_TERM }
+  );
+  if (!userId) return;
+  const { jobPostId, applicationStatus } = req.body;
+  const jobApplicationExists = await executeQuery(
```

Comment:
this is wrong. executeQuery always returns a truthy value. It returns an `{error: xx}` object in case of issues.

When writing code in the handler function (not helper function), its easiest to use the error handler.
```suggestion
  const results = await executeQueryAndEndSet500OnError(...0);
  if(!results) return; //early return because 500 would have been set.
  if(!results.length)  return res.status(200).send('Already applied');
```